File: NowCoder/nowcoder-90-L.py
import math
x = 1
a = 1
y = 1
b = 1
t = 1
t = input()
while(t):
	x,a,y,b = (int(i) for i in raw_input().split())
	ans1 = a*math.log(x)
	ans2 = b*math.log(y)
	if ans1 == ans2:
		print("Yes")
	else:
		print("No")
	t = t-1

File: NowCoder/nowcoder-90-L.cpp
#include <bits/stdc++.h>
using namespace std;
#define ine 0.00000001
int main(int argc, char const *argv[])
{
	int t;
	cin >> t;
	while(t--)
	{
		long long int a,b;
		double x,y;
		cin >> x >> a >> y >> b;
		double ans1 = a*log(x);
		double ans2 = b*log(y);
		if (ans1 - ans2 <= ine)
		{
			printf("Yes\n");
			continue;
		}else{
			printf("No\n");
		}	
	}
	return 0;
}